metals/src/main/scala/scala/meta/internal/metals/WorkspaceLspService.scala

```diff
@@ -848,10 +848,20 @@ class WorkspaceLspService(
                   }
               }(_ => true)
               .flatMap { mains =>
-                mains.headOption.fold(
+                mains.fold(
                   Future.failed[DebugSessionParams](
-                    DiscoveryFailures
-                      .NoMainClassFoundException(unresolvedParams.mainClass)
+                    Option(unresolvedParams.buildTarget)
+                      .map(buildTarget =>
+                        DiscoveryFailures
+                          .ClassNotFoundInBuildTargetException(
```

Comment:
This is actually the error message that is emitted deeper down but then get's lost on the way up. So I think it makes sense to emit it here.